server/core/runtime/plan_step_runner_test.go

```diff
@@ -630,6 +631,33 @@ func TestPlanStepRunner_TestRun_UsesConfiguredDistribution(t *testing.T) {
 
 }
 
+// Test custom regex on output method
+func TestCustomRegexFromPlanOutputFromPlanOutput(t *testing.T) {
+	cases := []struct {
+		in    string
+		out   string
+		regex string
+	}{
+		{
+			remotePlanOutput,
+			remotePlanOutput,
+			"",
+		},
+		{
+			remotePlanOutputSensitive,
+			remotePlanOutputSensitiveMasked,
+			`((?i)secret:\s")[^"]*`,
+		},
+	}
+
+	for _, c := range cases {
+		r, err := utils.ParseRegex(c.regex)
+		Ok(t, err)
+		output := runtime.FilterRegexFromPlanOutput(c.in, r)
```

Comment:
In this case instead of looping over the empty string you could just specify it as `nil` explicitly.
```suggestion
		regex *regexp.Regex
	}{
		{
			remotePlanOutput,
			remotePlanOutput,
			nil,
		},
		{
			remotePlanOutputSensitive,
			remotePlanOutputSensitiveMasked,
			regexp.MustCompile(`((?i)secret:\s")[^"]*`),
		},
	}

	for _, c := range cases {
		output := runtime.FilterRegexFromPlanOutput(c.in, c.regex)
```